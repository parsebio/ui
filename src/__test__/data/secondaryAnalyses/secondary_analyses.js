const mockAnalysisIds = {
  readyToLaunch: '868f4a31-da46-41a9-b18a-16b91a58302b',
  newProject: 'b936b0d2-0443-4a9c-884e-78da1f6002eb',
  latestTest: '06076ad1-c8c8-42fb-a425-974288a364a5',
};
const mockSecondaryAnalyses = [
  {
    id: mockAnalysisIds.latestTest,
    name: 'latest test',
    description: '',
    refGenome: null,
    createdAt: '2024-07-04 15:27:37.846038+00',
    kit: null,
    chemistryVersion: null,
    notifyByEmail: true,
    sampleNames: null,
    numOfSublibraries: null,
    parfileList: {
      start_timeout: 1000,
    },
  },
  {
    id: mockAnalysisIds.readyToLaunch,
    name: 'ready to launch',
    description: '',
    refGenome: 'Mmul10',
    createdAt: '2024-07-02 15:59:34.868158+00',
    kit: 'wt',
    chemistryVersion: '2',
    notifyByEmail: true,
    sampleNames: [
      'sample_1',
      'sample_2',
      'sample_3',
    ],
    numOfSublibraries: 2,
    parfileList: {
      start_timeout: 1000,
    },
  },
  {
    id: mockAnalysisIds.newProject,
    name: 'newProject',
    description: '',
    refGenome: null,
    createdAt: '2024-07-04 14:03:47.022592+00',
    kit: null,
    chemistryVersion: null,
    notifyByEmail: true,
    sampleNames: null,
    numOfSublibraries: null,
    parfileList: {
      start_timeout: 1000,
    },
  },
];

export {
  mockSecondaryAnalyses,
  mockAnalysisIds,
};
