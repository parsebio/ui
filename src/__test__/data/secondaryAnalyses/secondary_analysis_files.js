const mockAnalysisFiles = {
  files: [
    {
      id: '1000fe51-6a6d-4545-b717-a4c475849b94',
      name: 'S1_R1.fastq.gz',
      size: '70154428',
      type: 'wtFastq',
      upload: {
        status: 'uploaded',
      },
      createdAt: '2024-07-04 15:29:51.665893+00',
    },
    {
      id: 'd7ba4d40-e1ed-45a0-9eca-8f5caab0dd7b',
      name: 'S1_R2.fastq.gz',
      size: '55879599',
      type: 'wtFastq',
      upload: {
        status: 'uploaded',
      },
      createdAt: '2024-07-04 15:29:53.557915+00',
    },
    {
      id: 'c28efb10-6459-4107-b9ef-baf6b13968c4',
      name: 'S2_R1.fastq.gz',
      size: '70151802',
      type: 'wtFastq',
      upload: {
        status: 'uploaded',
      },
      createdAt: '2024-07-04 15:29:55.356064+00',
    },
    {
      id: '0740a094-f020-4c97-aceb-d0a708d0982e',
      name: 'S2_R2.fastq.gz',
      size: '55879479',
      type: 'wtFastq',
      upload: {
        status: 'uploaded',
      },
      createdAt: '2024-07-04 15:29:57.50098+00',
    },
    {
      id: 'a3e9f5f5-fbcb-46eb-be1f-bbeb9fe3491d',
      name: 'pbmc_1Mreads sltab.xlsm',
      size: '922034',
      type: 'samplelt',
      upload: {
        status: 'uploaded',
      },
      createdAt: '2024-07-08 14:17:49.164486+00',
    },
  ],
  pairMatches: [],
};

export default mockAnalysisFiles;
