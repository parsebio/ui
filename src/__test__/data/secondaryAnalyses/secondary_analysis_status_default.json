{
    "current": "not_created",
    "shouldRerun": true,
    "loading": false
}
