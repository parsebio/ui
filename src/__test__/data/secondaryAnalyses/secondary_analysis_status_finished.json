{
    "current": "finished",
    "loading": false,
    "error": false,
    "progress": {
        "running": 0,
        "succeeded": 17,
        "failed": 0
    },
    "tasksData": [
        {
            "process": "INQC",
            "status": "COMPLETED",
            "taskId": 1,
            "sublibrary": "s2_S2"
        },
        {
            "process": "INQC",
            "status": "COMPLETED",
            "taskId": 2,
            "sublibrary": "s1_S1"
        },
        {
            "process": "PRE",
            "status": "COMPLETED",
            "taskId": 3,
            "sublibrary": "s2_S2"
        },
        {
            "process": "PRE",
            "status": "COMPLETED",
            "taskId": 4,
            "sublibrary": "s1_S1"
        },
        {
            "process": "ALIGN",
            "status": "COMPLETED",
            "taskId": 5,
            "sublibrary": "s2_S2"
        },
        {
            "process": "ALIGN",
            "status": "COMPLETED",
            "taskId": 6,
            "sublibrary": "s1_S1"
        },
        {
            "process": "POST",
            "status": "COMPLETED",
            "taskId": 7,
            "sublibrary": "s2_S2"
        },
        {
            "process": "POST",
            "status": "COMPLETED",
            "taskId": 8,
            "sublibrary": "s1_S1"
        },
        {
            "process": "MOL",
            "status": "COMPLETED",
            "taskId": 9,
            "sublibrary": "s2_S2"
        },
        {
            "process": "MOL",
            "status": "COMPLETED",
            "taskId": 10,
            "sublibrary": "s1_S1"
        },
        {
            "process": "DGE",
            "status": "COMPLETED",
            "taskId": 11,
            "sublibrary": "s1_S1"
        },
        {
            "process": "DGE",
            "status": "COMPLETED",
            "taskId": 12,
            "sublibrary": "s2_S2"
        },
        {
            "process": "ANA",
            "status": "COMPLETED",
            "taskId": 13,
            "sublibrary": "s1_S1"
        },
        {
            "process": "ANA",
            "status": "COMPLETED",
            "taskId": 14,
            "sublibrary": "s2_S2"
        },
        {
            "process": "REP_CLEAN",
            "status": "COMPLETED",
            "taskId": 15,
            "sublibrary": "s1_S1"
        },
        {
            "process": "REP_CLEAN",
            "status": "COMPLETED",
            "taskId": 16,
            "sublibrary": "s2_S2"
        },
        {
            "process": "COMBINE",
            "status": "COMPLETED",
            "taskId": 17,
            "sublibrary": "s2_S2"
        },
        {
            "process": "COMBINE",
            "status": "COMPLETED",
            "taskId": 17,
            "sublibrary": "s1_S1"
        }
    ],
    "sublibraries": [
        "s2_S2",
        "s1_S1"
    ],
    "pipelineTasks": [
        "INQC",
        "PRE",
        "ALIGN",
        "POST",
        "MOL",
        "DGE",
        "ANA",
        "REP_CLEAN",
        "COMBINE"
    ],
    "totalTasks": 57,
    "shouldRerun": true
}
