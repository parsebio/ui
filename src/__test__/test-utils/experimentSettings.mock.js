import _ from 'lodash';

import initialState from 'redux/reducers/experimentSettings/initialState';
import { analysisTools } from 'utils/constants';

const sampleifiedConfig = (sampleIds, configToReplicate) => {
  const result = sampleIds.reduce(
    (acum, sampleId) => {
      // eslint-disable-next-line no-param-reassign
      acum[sampleId] = configToReplicate;
      return acum;
    },
    {},
  );

  return result;
};

const generateProcessingConfigMock = (sampleIds) => ({
  classifier: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      prefiltered: false,
      filterSettings: {
        FDR: 0.1,
      },
    }),
  },
  cellSizeDistribution: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        minCellSize: 10800,
        binStep: 200,
      },
    }),
  },
  mitochondrialContent: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        method: 'absoluteThreshold',
        methodSettings: {
          absoluteThreshold: {
            maxFraction: 0.1,
            binStep: 200,
          },
        },
      },
    }),
  },
  numGenesVsNumUmis: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        regressionType: 'gam',
        regressionTypeSettings: {
          gam: {
            'p.level': 0.00009,
          },
        },
      },
    }),
  },
  doubletScores: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        probabilityThreshold: 0.2,
        binStep: 0.05,
      },
    }),
  },
  dataIntegration: {
    enabled: true,
    analysisTool: analysisTools.SEURAT,
    dataIntegration: {
      method: 'harmony',
      methodSettings: {
        harmony: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        seuratv4: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        fastmnn: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        unisample: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
      },
    },
    dimensionalityReduction: {
      method: 'rpca',
      numPCs: 30,
      excludeGeneCategories: [],
    },
  },
  configureEmbedding: {
    enabled: true,
    embeddingSettings: {
      method: 'umap',
      useSaved: false,
      methodSettings: {
        umap: {
          minimumDistance: 0.1,
          distanceMetric: 'euclidean',
        },
        tsne: {
          perplexity: 30,
          learningRate: 200,
        },
      },
    },
    clusteringSettings: {
      method: 'louvain',
      methodSettings: {
        louvain: {
          resolution: 0.5,
        },
      },
    },
  },
});

const generateExperimentSettingsMock = (sampleIds) => {
  const mockedProcessingConfig = {
    ...initialState.processing,
    ...generateProcessingConfigMock(sampleIds),
  };

  return {
    ...initialState,
    processing: mockedProcessingConfig,
    originalProcessing: _.cloneDeep(mockedProcessingConfig),
    info: {
      ...initialState.info,
      pipelineVersion: 1,
    },
  };
};

export default generateExperimentSettingsMock;
