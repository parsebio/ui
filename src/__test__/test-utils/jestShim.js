/**
 * This file is required for jest to work with '@cfaester/enzyme-adapter-react-18'
 * module.
 */
import {
  TextDecoder, TextEncoder,
} from 'util';

global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;
