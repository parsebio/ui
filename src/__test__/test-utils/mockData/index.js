import _ from 'lodash';
import { analysisTools } from 'utils/constants';
import generateMockSamples from './generateMockSamples';
import generateMockExperiments from './generateMockExperiments';
import generateMockProcessingConfig from './generateMockProcessingConfig';

// This file is a work in progress to generate mock data for the endpoints of the test.
// The generated data should be usable as response data to be sent via the mocked API in mockAPI.js
// The module should also expose the generated data that can be imported and used easily in tests

const responseData = {};

responseData.experiments = generateMockExperiments(2);

responseData.samples = [generateMockSamples(
  responseData.experiments[0],
  3,
)];

responseData.processingConfigSeurat = generateMockProcessingConfig(analysisTools.SEURAT, 3);
responseData.processingConfigScanpy = generateMockProcessingConfig(analysisTools.SCANPY, 3);

// Add samples to first experiment
const samples = responseData.samples[0];
const sampleIds = _.map(samples, 'id');
responseData.experiments[0].samplesOrder = sampleIds;

const { experiments, processingConfigSeurat, processingConfigScanpy } = responseData;

export {
  experiments,
  samples,
  processingConfigSeurat,
  processingConfigScanpy,
  responseData,
};
