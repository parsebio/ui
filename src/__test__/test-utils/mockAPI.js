import _ from 'lodash';
import backendStatusData from '__test__/data/backend_status.json';
import processingConfigData from '__test__/data/processing_config.json';
import mockDemoExperiments from '__test__/test-utils/mockData/mockDemoExperiments.json';
import { mockSecondaryAnalyses } from '__test__/data/secondaryAnalyses/secondary_analyses';
import mockSecondaryAnalysisStatusDefault from '__test__/data/secondaryAnalyses/secondary_analysis_status_default.json';
import fake from '__test__/test-utils/constants';
import mockAnalysisFiles from '__test__/data/secondaryAnalyses/secondary_analysis_files';
import downloadFromS3 from 'utils/work/downloadFromS3';

import {
  responseData,
} from '__test__/test-utils/mockData';
import { analysisTools } from 'const';

const cellSetsData = require('__test__/data/cell_sets.json');

const setupDownloadCellSetsFromS3Mock = (customCellSets = cellSetsData) => {
  downloadFromS3.mockImplementation((resource, signedUrl) => {
    // eslint-disable-next-line global-require
    if (signedUrl === 'mock-cellsets-signed-url') {
      return Promise.resolve(customCellSets);
    }
    return Promise.reject(new Error('Invalid signed URL'));
  });
};

const promiseResponse = (response, options = {}) => (
  Promise.resolve(new Response(response, options)));

const statusResponse = (code, body) => (
  Promise.resolve({
    status: code,
    body: JSON.stringify(body),
  })
);

const delayedResponse = (response, delay = 10000, options = {}) => new Promise((resolve) => {
  setTimeout(() => {
    resolve(new Response(response, options));
  }, delay);
});

const workerDataResult = (data) => Promise.resolve(_.cloneDeep(data));

const fetchWorkMock = (mockedResults) => ((experimentId, body) => {
  if (typeof mockedResults[body.name] === 'function') {
    return mockedResults[body.name]();
  }
  return workerDataResult(mockedResults[experimentId]);
});

const generateDefaultMockAPIResponses = (projectId, analysisTool = analysisTools.SEURAT) => ({
  [`experiments/${projectId}$`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === projectId)),
  ),
  [`/v2/experiments/${projectId}$`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === projectId)),
  ),
  [`experiments/${projectId}/processingConfig$`]: () => {
    const processingConfigDataClone = _.cloneDeep(processingConfigData);
    processingConfigDataClone.processingConfig.dataIntegration.analysisTool = analysisTool;

    return promiseResponse(
      JSON.stringify(processingConfigDataClone),
    );
  },
  [`experiments/${projectId}/cellSets$`]: () => {
    setupDownloadCellSetsFromS3Mock();
    return promiseResponse(
      JSON.stringify('mock-cellsets-signed-url'),
    );
  },
  [`experiments/${projectId}/backendStatus$`]: () => promiseResponse(
    JSON.stringify(backendStatusData),
  ),
  experiments$: () => promiseResponse(
    JSON.stringify(responseData.experiments),
  ),
  [`experiments/${projectId}/samples$`]: () => promiseResponse(
    JSON.stringify(responseData.samples[0]),
  ),
  '/v2/experiments/examples$': () => promiseResponse(
    JSON.stringify(mockDemoExperiments),
  ),
  'experiments/clone$': () => promiseResponse(
    JSON.stringify(fake.CLONED_EXPERIMENT_ID),
  ),
  [`/v2/workRequest/${projectId}`]: () => statusResponse(200, 'OK'),
  '/v2/secondaryAnalysis$': () => {
    const mockedMainSecondary = _.cloneDeep(mockSecondaryAnalyses[1]);
    mockedMainSecondary.id = projectId;

    return promiseResponse(
      JSON.stringify([
        mockedMainSecondary,
        ...mockSecondaryAnalyses,
      ]),
    );
  },
  [`/v2/secondaryAnalysis/${projectId}/executionStatus`]: () => promiseResponse(
    JSON.stringify(mockSecondaryAnalysisStatusDefault),
  ),
  [`/v2/secondaryAnalysis/${projectId}/files`]: (req) => {
    if (req.method === 'GET') {
      return promiseResponse(
        JSON.stringify(mockAnalysisFiles),
      );
    }

    if (req.method === 'POST') {
      return promiseResponse(
        JSON.stringify({ fileId: 'mockFileId' }),
      );
    }
  },
  [`/v2/projects/${projectId}/upload/.*/part/\\d+/signedUrl$`]: (req) => {
    const partNumber = req.url.match(/part\/(\d+)\/signedUrl/)[1];
    return promiseResponse(JSON.stringify(`mockSignedUrl${partNumber}`));
  },
  '/v2/cliUpload/token$': () => promiseResponse(JSON.stringify('token')),
});

const mockAPI = (apiMapping) => (req) => {
  const path = req.url;

  const key = _.find(Object.keys(apiMapping), (matcherStr) => new RegExp(matcherStr).test(path));

  if (!key) {
    return statusResponse({
      status: 404,
      body: `Path ${path} is undefined`,
    });
  }

  return apiMapping[key](req);
};

export default mockAPI;
export {
  generateDefaultMockAPIResponses,
  promiseResponse,
  statusResponse,
  delayedResponse,
  workerDataResult,
  fetchWorkMock,
  setupDownloadCellSetsFromS3Mock,
};
