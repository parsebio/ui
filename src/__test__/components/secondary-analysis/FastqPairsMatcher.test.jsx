import React from 'react';
import {
  render, screen, fireEvent, within,
} from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import { Provider } from 'react-redux';
import { makeStore } from 'redux/store';
import {
  loadSecondaryAnalyses,
  loadSecondaryAnalysisFiles,
  loadSecondaryAnalysisStatus,
  setActiveSecondaryAnalysis,
} from 'redux/actions/secondaryAnalyses';
import mockAPI, {
  generateDefaultMockAPIResponses,
  promiseResponse,
} from '__test__/test-utils/mockAPI';
import { mockAnalysisIds } from '__test__/data/secondaryAnalyses/secondary_analyses';

import FastqPairsMatcher from 'components/secondary-analysis/FastqPairsMatcher';
import mockAnalysisFilesWithImmune from '__test__/data/secondaryAnalyses/secondary_analysis_files_with_immune';

const readyToLaunchResponses = generateDefaultMockAPIResponses(mockAnalysisIds.readyToLaunch);
const emptyAnalysisResponses = generateDefaultMockAPIResponses(mockAnalysisIds.emptyAnalysis);
const mockAPIResponses = {
  ...emptyAnalysisResponses,
  ...readyToLaunchResponses,
  [`/v2/secondaryAnalysis/${mockAnalysisIds.readyToLaunch}/files`]: () => promiseResponse(
    JSON.stringify(mockAnalysisFilesWithImmune),
  ),
};

enableFetchMocks();

const openAndSelect = () => {

};

describe('FastqPairsMatcher', () => {
  let storeState;

  const renderSelect = async () => await render(
    <Provider store={storeState}>
      <FastqPairsMatcher />
    </Provider>,
  );

  beforeEach(async () => {
    jest.clearAllMocks();

    fetchMock.resetMocks();
    fetchMock.mockIf(/.*/, mockAPI(mockAPIResponses));
    storeState = makeStore();
    await storeState.dispatch(loadSecondaryAnalyses());
    await storeState.dispatch(loadSecondaryAnalysisFiles(mockAnalysisIds.readyToLaunch));
    await storeState.dispatch(loadSecondaryAnalysisStatus(mockAnalysisIds.readyToLaunch));

    await act(async () => {
      await storeState.dispatch(setActiveSecondaryAnalysis(mockAnalysisIds.readyToLaunch));
    });
  });

  it('Renders correctly', async () => {
    await renderSelect();
    // Correct sublibrary numbers
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('2')).toBeInTheDocument();

    // Correct fastq sublibrary numbers
    expect(screen.getByText('S1')).toBeInTheDocument();
    expect(screen.getByText('S2')).toBeInTheDocument();

    // Two empty selects for immune pairs
    expect(screen.getAllByText('Select immune pair')).toHaveLength(2);
  });

  it('Selecting functionality works fine', async () => {
    await renderSelect();

    // Not selected first
    expect(document.querySelector('.ant-select-selection-item[title="TCR_S1"]')).not.toBeInTheDocument();
    expect(document.querySelector('.ant-select-selection-item[title="TCR_S2"]')).not.toBeInTheDocument();

    const selects = screen.getAllByRole('combobox');

    // Open the first immune select dropdown and select the first immune pair
    fireEvent.mouseDown(selects[0]);
    const tcrS1Option = screen.getAllByText('TCR_S1')[1];

    await act(() => {
      fireEvent.click(tcrS1Option);
    });

    expect(document.querySelector('.ant-select-selection-item[title="TCR_S1"]')).toBeInTheDocument();

    // Open the second immune select dropdown and select the second immune pair
    fireEvent.mouseDown(selects[1]);
    const tcrS2Option = screen.getAllByText('TCR_S2')[1];

    await act(() => {
      fireEvent.click(tcrS2Option);
    });

    expect(document.querySelector('.ant-select-selection-item[title="TCR_S2"]')).toBeInTheDocument();

    // Check that updatePairMatch's URL was called for each selection
    const calls = fetchMock.mock.calls.filter(([url]) => (
      url.includes(`${mockAnalysisIds.readyToLaunch}/files/pairMatches`)
    ));

    expect(calls).toHaveLength(2);

    expect(calls[0]).toMatchSnapshot();
    expect(calls[1]).toMatchSnapshot();
  });
});
