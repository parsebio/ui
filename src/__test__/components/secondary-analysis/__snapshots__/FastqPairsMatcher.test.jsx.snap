// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FastqPairsMatcher Selecting functionality works fine 1`] = `
[
  "http://localhost:3000/v2/secondaryAnalysis/868f4a31-da46-41a9-b18a-16b91a58302b/files/pairMatches",
  {
    "body": "[{"wtR1FileId":"1000fe51-6a6d-4545-b717-a4c475849b94","wtR2FileId":"d7ba4d40-e1ed-45a0-9eca-8f5caab0dd7b","immuneFileR1Id":"1000fe51-6a6d-4545-b717-a4c475849b95","immuneFileR2Id":"d7ba4d40-e1ed-45a0-9eca-8f5caab0dd7c"}]",
    "headers": {
      "Content-Type": "application/json",
    },
    "method": "POST",
  },
]
`;

exports[`FastqPairsMatcher Selecting functionality works fine 2`] = `
[
  "http://localhost:3000/v2/secondaryAnalysis/868f4a31-da46-41a9-b18a-16b91a58302b/files/pairMatches",
  {
    "body": "[{"wtR1FileId":"1000fe51-6a6d-4545-b717-a4c475849b94","wtR2FileId":"d7ba4d40-e1ed-45a0-9eca-8f5caab0dd7b","immuneFileR1Id":"c28efb10-6459-4107-b9ef-baf6b13968c5","immuneFileR2Id":"0740a094-f020-4c97-aceb-d0a708d0982f"}]",
    "headers": {
      "Content-Type": "application/json",
    },
    "method": "POST",
  },
]
`;
